deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LSTMHelpers.java

```diff
@@ -168,6 +170,15 @@ static public FwdPassReturn activateHelper(final Layer layer, final NeuralNetCon
             prevOutputActivations = Nd4j.zeros(new int[] {miniBatchSize, hiddenLayerSize});
         }
 
+        if (helper != null && Nd4j.dataType() != DataBuffer.Type.HALF) {
```

Comment:
ActivationSigmoid and ActivationTanH: these should be checked at this point too I think - before we get to CudnnLSTMHelper, so we can fall back to our implementation if we have non sigmoid or non tanh activations.
Or: we add a method like activationsSupported(IActivation gates, IActivation afn) on the lstm helper interface.